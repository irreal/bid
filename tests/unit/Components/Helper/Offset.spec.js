import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import Offset from "@/components/Helper/Offset";
import Vuetify from "vuetify";
import Vue from "vue";

const localVue = createLocalVue();
localVue.use(Vuex);
Vue.use(Vuetify);

describe("Offset helper", () => {
  it("renders the component", () => {
    // arrange
    const oh = shallowMount(Offset);
    // assert
    expect(oh.html()).toMatchSnapshot();
  });

  it("renders the component with slots", () => {
    // arrange
    const oh = shallowMount(Offset, {
      slots: {
        default: "<p>this content is offset</p>"
      }
    });
    // assert
    expect(oh.html()).toMatchSnapshot();
    expect(oh.html()).toContain("this content is offset");
  });
});
