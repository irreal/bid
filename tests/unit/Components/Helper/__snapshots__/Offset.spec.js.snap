// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offset helper renders the component 1`] = `<div class="v-offset" style="top: -0px; margin-bottom: -0px;"></div>`;

exports[`Offset helper renders the component with slots 1`] = `
<div class="v-offset" style="top: -0px; margin-bottom: -0px;">
  <p>this content is offset</p>
</div>
`;
